### PR TITLE
Issue/3884 descriptive hpc job name (#1)

### PR DESCRIPTION
**Pull request by kannon92**
## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

A fix for issue 3884.  The goal of this PR is to give more detailed names for jobnames in HPC schedulers.  We would like to be able to check the status of a step of a workflow and/or delete a workflow.

## Reviewer Checklist

 * [x] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [x] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [x] Read through the code changes. Make sure that it doesn't have:
    * [x] Addition of trailing whitespace.
    * [x] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [x] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [x] New functions or classes without informative docstrings.
    * [x] Changes to semantics not reflected in the relevant docstrings.
    * [x] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [x] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [x] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: src/toil/cwl/cwltoil.py [hunk 1]
```diff
@@ -3381,11 +3388,13 @@ def main(args: Union[List[str]] = None, stdout: TextIO = sys.stdout) -> int:
                 # were required.
                 rm_unprocessed_secondary_files(param_value)
 
+            logger.debug('tool ', tool)
             try:
                 wf1, _ = makeJob(
                     tool=tool,
                     jobobj={},
                     runtime_context=runtime_context,
+                    workflow_name='workflow',
```
Comment (kannon92) [inline src/toil/cwl/cwltoil.py:3397]: I'm not sure if I can get the workflow name from the ID at this step.

**Review by adamnovak — APPROVED** [review]
This looks pretty good. It might make sense to leave workflow_name unset if we can't actually get it, but this is an improvement over what we have now.

**Comment by adamnovak** [comment]
I pulled this in for CI testing as `issues/3884-descriptive-hpc-jobName`.

**Comment by kannon92** [comment]
What should we do about the CI failure?  It looks like a known failure with K8 test.  I tried updating my fork to the latest and pushing.  But I guess that didn't fix it?

File: src/toil/cwl/cwltoil.py (hunk 1 above)
Comment (mr-c) [reply src/toil/cwl/cwltoil.py:3434]: How about `tool.tool['id']`  ?

**Review by adamnovak — APPROVED** [review]
I'm going to approve and also close this; I glommed it into #3941 to make the minor edits it needs to actually merge.

**Pull request closed without merging**